### ISSUE TITLE
BIM Update error: Unexpected Error TMSavePoint line 1380

### ISSUE THREAD

**Issue by tpcorcoran** [microsoft/Analysis-Services#145]
Hello all - I'll start with the error and then add context - thank you for any help...

**Error**: We have a directquery model that was deployed to AS (PBI Premium).  We then made some schema changes to tables on this model, as well as added some measures.  I'm running into the same error whether I deploy it using Tabular Editor or update it using ALM Toolkit...

> An unexpected error occurred (file 'tmsavepoint.cpp', line 1380, function 'TMSavePoint::ThrowObjectNotFoundError').
> 
> Technical Details:
> RootActivityId: 7e7fde9e-73c9-4f7f-a510-89fbf15ed54f
> Date (UTC): 5/11/2022 8:02:41 PM

**Context**: We have a master (enterprise) dataset that uses an import connection back to Databricks.  We then build smaller (report-level) datasets with directquery connections back to the enterprise ds to pull in a subset of tables.  The above changes were first made to the master dataset, with those schema changes deployed successfully using ALM.  I then used the locally stored PBI file for the report-level dataset to pull in those table/measure changes, then saved off that bim file to use for the aforementioned failed deployment.  

One final note - I am able to successfully deploy the dataset using Tabular Editor if I save it as a new file name (rather than versioning up the existing model)

Any assistance would be appreciated or let me know if more info is needed.  Thanks
(Even if we can get a more specific message on this error than that would be awesome)

**Comment by tpcorcoran** [comment]
I still open to thoughts, but might have figured this out.  Turns out the two models had different Shared Expression definitions that are used in each table's partition definition.  I ran a test by redeploying the original version with a change to that Expression and then used ALM to update it and that was successful.  

I'll leave this open for now, but can close if this sounds like a logical explanation
